src/monmove.cpp

```diff
@@ -1091,51 +1091,64 @@ void monster::knock_back_from(int x, int y)
  */
 bool monster::will_reach(int x, int y)
 {
- monster_attitude att = attitude(&(g->u));
- if (att != MATT_FOLLOW && att != MATT_ATTACK && att != MATT_FRIEND && att != MATT_ZLAVE)
-  return false;
+    monster_attitude att = attitude( &(g->u) );
+    if( att != MATT_FOLLOW && att != MATT_ATTACK && att != MATT_FRIEND && att != MATT_ZLAVE ) {
+        return false;
+    }
 
- if (has_flag(MF_DIGS))
-  return false;
+    if( has_flag(MF_DIGS) || has_flag(MF_AQUATIC) ) {
+        return false;
+    }
 
- if (has_flag(MF_IMMOBILE) && (posx() != x || posy() != y))
-  return false;
+    if( has_flag(MF_IMMOBILE) && ( posx() != x || posy() != y ) ) {
+        return false;
+    }
 
- std::vector<point> path = g->m.route( posx(), posy(), x, y, 0 );
- if (path.empty())
-   return false;
+    std::vector<point> path = g->m.route( posx(), posy(), x, y, 0 );
+    if( path.empty() ) {
+        return false;
+    }
 
- if (has_flag(MF_SMELLS) && g->scent(posx(), posy()) > 0 &&
-     g->scent(x, y) > g->scent(posx(), posy()))
-  return true;
+    if( has_flag(MF_SMELLS) && g->scent( posx(), posy() ) > 0 &&
+        g->scent(x, y) > g->scent( posx(), posy() ) ) {
+        return true;
+    }
 
- if (can_hear() && wandf > 0 && rl_dist(wandx, wandy, x, y) <= 2 &&
-     rl_dist(posx(), posy(), wandx, wandy) <= wandf)
-  return true;
+    if( can_hear() && wandf > 0 && rl_dist( wandx, wandy, x, y ) <= 2 &&
+        rl_dist( posx(), posy(), wandx, wandy ) <= wandf ) {
+        return true;
+    }
 
- int t;
- if (can_see() && g->m.sees(posx(), posy(), x, y, g->light_level(), t))
-  return true;
+    int t;
+    if( can_see() && g->m.sees( posx(), posy(), x, y, g->light_level(), t ) ) {
+        return true;
+    }
 
- return false;
+    return false;
 }
 
 int monster::turns_to_reach(int x, int y)
 {
+    // Note: values in this function aren't turns, but "pairs of turns"
+    // This is because move_cost returns 2 for flat ground, 8 for rubble etc.
+    // This function is a(n old) temporary hack that should soon be removed
```

Comment:
What pairs? I don't see any pairs in this function. `monster::move_to` decreases the moves of the monster by the value returns by `calc_movecost`.